test-vectors/single_message.json

```diff
@@ -0,0 +1,32 @@
+{
+  "class": "messages",
+  "selector": "<some predicate for the driver to determine if this vector applies to the implementation, e.g. to deal with protocol upgrades>",
+
+  "_meta": {
+    "id": "test_vector_id",
+    "version": "version_id",
+    "gen": {
+      "source": "lotus",
+      "version": "0.4.1+git.27d74337+api0.8.1"
+    }
+  },
+
+  "preconditions": {
+    "state_tree": {
+      "t01": "0x0123456789abcdef",
+      "t02": "0x0123456789abcdef",
+      "t03": "0x0123456789abcdef"
+    }
+  },
+
+  "apply_messages": [
+    "0x89004300ba20583103b85358cc1c968d826c0a0efcde99b6e4db138edfc051f188a4471cfc23ccf7884adbbac912e45f7363774bf16f83ca8f1902b44800505e1442e66b744200011a09d225c80758c68219016f58c099a447a8c80988e1c061dd3a8dd674033c3a6083c4b55cf035e6e8de3baf09436ad00adf6eb14d58ae6691d22ce82255874047e7aa0b9a86bbf424afc6d89dc29f7b253dd04c88a625b86345f05aa0bdee503971c1e9735ed5b81c31ad41a1f10b387d96151d1cbd0c4286267daf04087fd8d5272f870994c3732f3fc4f54287a485d6c8dafb6056df3fc9233f849027b4c1272850435126fef65a8363768a858c0cc5f9d9114ff378ad50cab25ab0a2f685ca63d9f1f7727da4d06db5b80b5f"
+  ],
+
+  "postconditions": {
+    "state_tree": {
+      "root_cid": "bafy2bzacecfk5enlkya62353pwt5wjn3v7mldibtzgohfpbjw4xmabm34evz4",
+      "car_bytes": "0x890043009920583103b5b1fab6769fcb464146d192af785915d8e1dacff264b2be3bcb7bf4064a77cb3b13e27063ea4825dc48cd0a7df77e211916e748002740f0c493a2824200011a09d9c4dc0758c68219091c58c083e7c59e748e070b382a813f4ffe8ead2a090469aaa37f0664b82c092bc78b43fac300335cd2d6e4a8c71ce138003c0fa8ee1707b41aa87f35c811613d59ac53449aef8263aa51c876f2e67c185118ec9628f73442f58e093909ce952e431e770ac693cdbe99f2f9404ad3ca754196ab443c6c66afa41310e705e453496bdcedf0f271529dc7374021d7b80d343397f996d06bf11e753a49eaed7b3ce80a8e9bbaa48bf83223527bd8a2933f002949fdb8058103336a4a979375d5e963cdd7b1"
+    }
+  }
+}
```

Comment:
In the event that this tests fails (meaning the resulting state root != `root_cid`(?)) what information will be given to the caller of the test? State roots can differ for a variety of reasons: different gas usage, exit codes, return values, etc. I think it will be helpful to surface why state roots are different -- is tooling being built to facilitate that?

Reply:
https://github.com/filecoin-project/oni/issues/166